core/src/main/java/hudson/tasks/Fingerprinter.java

```diff
@@ -92,13 +93,40 @@
      */
     private final String targets;
 
+    /**
+     * Default null 'excludes' pattern as in Ant.
+     */
+    private String excludes = null;
+
+    /**
+     * Default ant exclusion
+     */
+    private Boolean defaultExcludes = true;
+
+    /**
+     * Indicate whether include and exclude patterns should be considered as case sensitive
+     */
+    private Boolean caseSensitive = true;
+
     @Deprecated
     Boolean recordBuildArtifacts;
 
     @DataBoundConstructor public Fingerprinter(String targets) {
         this.targets = targets;
     }
 
+    @DataBoundSetter public void setExcludes(String excludes) {
+        this.excludes = excludes;
```

Comment:
Should use `Util.fixEmpty`. Otherwise there will be a mismatch between `""` as saved by the GUI form and `null` as returned by a default instance. This will result in a gratuitous

```groovy
excludes: ''
```

added in the **Snippet Generator** for Pipeline. For freestyle projects, will result in a gratuitous

```xml
<excludes></excludes>
```